pkg/component/component.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
this needs to be updated with component.go from blakes PR.
this has a lot of dumb logic and not implemented handlers which blake already solved so i will wait for him to push changes in